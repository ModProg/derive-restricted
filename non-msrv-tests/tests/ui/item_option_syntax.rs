use std::marker::PhantomData;

use derive_where::DeriveWhere;

#[derive(DeriveWhere)]
// Rust itself already fails to parse this and will provide a separate error message.
#[derive_where = invalid]
struct InvalidAttribute<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where = "invalid"]
struct WrongAttributeSyntax<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where()]
struct EmptyAttribute<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(Debug = "option"; T)]
struct WrongOptionSyntax<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(Debug(); T)]
struct EmptyOption<T, U>(T, PhantomData<U>);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Debug(option); T)]
struct UnsupportedOption<T, U>(T, PhantomData<U>);

fn main() {}
