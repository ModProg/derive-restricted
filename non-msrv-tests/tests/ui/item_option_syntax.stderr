error: expected parentheses: #[derive_where(...)]
 --> tests/ui/item_option_syntax.rs:7:16
  |
7 | #[derive_where = invalid]
  |                ^

error: unexpected option syntax
  --> tests/ui/item_option_syntax.rs:11:3
   |
11 | #[derive_where = "invalid"]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^

error: empty `derive_where` found
  --> tests/ui/item_option_syntax.rs:15:3
   |
15 | #[derive_where()]
   |   ^^^^^^^^^^^^^^

error: unexpected option syntax
  --> tests/ui/item_option_syntax.rs:19:16
   |
19 | #[derive_where(Debug = "option"; T)]
   |                ^^^^^^^^^^^^^^^^

error: empty attribute option found
  --> tests/ui/item_option_syntax.rs:23:16
   |
23 | #[derive_where(Debug(); T)]
   |                ^^^^^^^

error: `Debug` doesn't support any options
  --> tests/ui/item_option_syntax.rs:27:16
   |
27 | #[derive_where(Debug(option); T)]
   |                ^^^^^^^^^^^^^

error: unexpected token: `invalid`
 --> tests/ui/item_option_syntax.rs:7:18
  |
7 | #[derive_where = invalid]
  |                  ^^^^^^^
